miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleRestController.java

```diff
@@ -203,6 +203,42 @@ public DataTablesResponseDto<SampleDto> getDTArrayedSamplesByProject(@PathVariab
         PaginationFilter.arrayed(true));
   }
 
+  @GetMapping(value = "/dt/samples-prepared/{id}", produces = {"application/json"})
+  @ResponseBody
+  public DataTablesResponseDto<SampleDto> getPreparedSamplessByRequisition(@PathVariable("id") long id,
+      HttpServletRequest request) throws IOException {
+    List<Sample> requisitionedSamples = sampleService.list(0, 0, false, "id", PaginationFilter.requisitionId(id));
+    List<Sample> supplementalSamples =
+        sampleService.list(0, 0, false, "id", PaginationFilter.supplementalToRequisitionId(id));
+    Set<Long> rootSampleIds = Stream.concat(requisitionedSamples.stream(), supplementalSamples.stream())
+        .map(Sample::getId)
+        .collect(Collectors.toSet());
+    Set<Long> allDerivativeSampleIds = new HashSet<>();
+    List<Sample> tissueSamples = sampleService.getChildren(rootSampleIds, SampleTissue.CATEGORY_NAME, id);
+    Set<Long> tissueSampleIds = tissueSamples.stream()
+        .map(Sample::getId)
+        .collect(Collectors.toSet());
+    allDerivativeSampleIds.addAll(tissueSampleIds);
+    List<Sample> tissueProcessingSamples = sampleService.getChildren(
+        tissueSampleIds, SampleTissueProcessing.CATEGORY_NAME, id);
+    Set<Long> tissueProcessingIds = tissueProcessingSamples.stream()
+        .map(Sample::getId)
+        .collect(Collectors.toSet());
+    allDerivativeSampleIds.addAll(tissueProcessingIds);
+    List<Sample> stockSamples = sampleService.getChildren(rootSampleIds, SampleStock.CATEGORY_NAME, id);
+    Set<Long> stockSampleIds = stockSamples.stream()
+        .map(Sample::getId)
+        .collect(Collectors.toSet());
+    allDerivativeSampleIds.addAll(stockSampleIds);
+    List<Sample> aliquotSamples = sampleService.getChildren(stockSampleIds, SampleAliquot.CATEGORY_NAME, id);
```

Comment:
Sorry, but my suggestion you've implemented to do the lookup from the previous level isn't actually going to work. It's possible that there will be no stock children (e.g. if stocks were requisitioned), so then this wouldn't find the aliquots. Similar problem for the other levels too. All will need to use `rootSampleIds`.

It would be better to have a single service level method that gets all the samples - or even just their IDs - but I think this is reasonably fast, so we don't need to do that for now.